connectors/sources/sharepoint_online.py

```diff
@@ -626,20 +626,82 @@ async def site_users(self, site_web_url):
         except NotFound:
             return
 
-    async def sites(self, parent_site_id, allowed_root_sites):
-        select = ""
+    async def sites(
+        self,
+        sharepoint_host,
+        allowed_root_sites,
+        enumerate_all_sites=True,
+        fetch_subsites=False,
+    ):
+        if allowed_root_sites == [WILDCARD] or enumerate_all_sites:
```

Comment:
I'd suggest to make it consistent by checking

```python
WILDCARD in allowed_root_sites
```

This can work with the edge case when customers configure something like:
```
Sean'steamsite, *, Chenhui'scommunicationsite
```

Reply:
I'd assumed that was a logical bug. Why would you configure `*` AND anything else? The star covers all other sites, right?

Reply:
See https://elastic.slack.com/archives/C01795T48LQ/p1695396448194609?thread_ts=1695395634.669799&cid=C01795T48LQ

I'm going to resolve this for now, but we can open up a follow-up issue if you still have concerns.